spectator-ext-stackdriver/src/main/java/com/netflix/spectator/stackdriver/StackdriverWriter.java

```diff
@@ -0,0 +1,760 @@
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.stackdriver;
+
+
+import com.netflix.spectator.controllers.MetricsController;
+
+import com.google.api.services.monitoring.v3.Monitoring;
+import com.google.api.services.monitoring.v3.MonitoringScopes;
+import com.google.api.services.monitoring.v3.model.CreateTimeSeriesRequest;
+import com.google.api.services.monitoring.v3.model.LabelDescriptor;
+import com.google.api.services.monitoring.v3.model.ListMetricDescriptorsResponse;
+import com.google.api.services.monitoring.v3.model.Metric;
+import com.google.api.services.monitoring.v3.model.MetricDescriptor;
+import com.google.api.services.monitoring.v3.model.Point;
+import com.google.api.services.monitoring.v3.model.TimeInterval;
+import com.google.api.services.monitoring.v3.model.TimeSeries;
+import com.google.api.services.monitoring.v3.model.TypedValue;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.common.collect.Lists;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Meter;
+import com.netflix.spectator.api.Measurement;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Tag;
+import com.netflix.spectator.controllers.filter.MeasurementFilter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/* These are builtin metric types that we might be able to use
+ * except probably not because there isnt a label to say this is our use
+ * as opposed to the assumption that "there is only one" of them.
+
+    "agent.googleapis.com/jvm/memory/usage"
+    "agent.googleapis.com/jvm/cpu/time"
+    "agent.googleapis.com/jvm/gc/time"
+    "agent.googleapis.com/jvm/gc/time"
+    "agent.googleapis.com/jvm/thread/num_live"
+    "agent.googleapis.com/jvm/thread/peak"
+    "agent.googleapis.com/jvm/uptime"
+
+    "agent.googleapis.com/tomcat/manager/sessions"
+    "agent.googleapis.com/tomcat/request_processor/error_count"
+    "agent.googleapis.com/tomcat/request_processor/processing_time"
+    "agent.googleapis.com/tomcat/request_processor/request_count"
+    "agent.googleapis.com/tomcat/request_processor/traffic_count"
+    "agent.googleapis.com/tomcat/threads/current"
+    "agent.googleapis.com/tomcat/threads/busy"
+*/
```

Comment:
This isnt intended for javadoc (single star). It's more an informal note for future (not so distant) reference.